### ISSUE TITLE
Cannot add CollapseBox within Box component

### ISSUE THREAD

**Issue by kyuhas** [open-olive/loop-development-kit#468]
Hi! Right now, you cannot add a collapse box component within a box component. I've provided a simple code example in this issue (see below).

```
import { Direction, JustifyContent, StyleSize } from '@oliveai/ldk/dist/whisper';
import React from 'react';

export const TestCollapseBoxWithinBox = () => {
  return (
    <oh-box
      direction={Direction.Vertical}
      justifyContent={JustifyContent.Left}
      layout={{ marginTop: StyleSize.Small, marginBottom: StyleSize.Small }}
    >
      <oh-message body="some value here" />
      <oh-collapse-box open={false} label="More">
        <oh-message body="another value here" />
      </oh-collapse-box>
    </oh-box>
  );
};
```

When I build and run my loop, the collapse box does not render (see image below) and the following appears in the logs: `[2021-12-15 11:14:03.000] [desktop-core] [warning] Cannot add item of type collapseBox to parent of type box loop=f55098eb-7bef-4215-af8e-180f4468720d`.
<img width="368" alt="Screen Shot 2021-12-15 at 11 14 14 AM" src="https://user-images.githubusercontent.com/26147799/146223824-8dbd0853-36a9-46a8-823f-3e990314cc87.png">

Is it possible to remove the limitation of not adding a collapse box within a box component? To get around this, I have been using fragments, but sometimes if I have a collapse box deeply nested with a bunch of components, I do need those outer boxes to help with formatting.

**Comment by MichaelOliveMcCarrick** [comment]
I know we've looked into nested collapse boxes in the past and it was ultimately decided to not implement them at the time. What troubles are you having with the formatting, we can look into having a single collapse box nested, but there also may be other improvements we can make down the line to provide a better experience.

**Comment by kyuhas** [comment]
My actual code is much more complex than this, but as a simple example, suppose I want to center align a bunch of items on a page, but then also have a collapse box in the middle of the items (see code below). 

```
import { AlignItems, Direction, JustifyContent, StyleSize } from '@oliveai/ldk/dist/whisper';
import React from 'react';

export const TestCollapseBoxWithinBox = () => {
  return (
    <oh-box
      direction={Direction.Vertical}
      justifyContent={JustifyContent.Left}
      alignItems={AlignItems.Center}
      layout={{ marginTop: StyleSize.Medium, marginBottom: StyleSize.Medium }}
    >
      <oh-message body="A" />
      <oh-message body="B" />
      <oh-collapse-box open={false} label="More">
        <oh-message body="C" />
      </oh-collapse-box>
      <oh-message body="D" />
      <oh-message body="E" />
    </oh-box>
  );
};
```

Right now, the collapse box will not appear. The only way to get it to show up, while also centering the other items is to do the following:
```
/* eslint-disable react/jsx-fragments */
import { AlignItems, Direction, JustifyContent, StyleSize } from '@oliveai/ldk/dist/whisper';
import React from 'react';

export const TestCollapseBoxWithinBox = () => {
  return (
    <>
      <oh-box
        direction={Direction.Vertical}
        justifyContent={JustifyContent.Left}
        alignItems={AlignItems.Center}
        layout={{ marginTop: StyleSize.Medium, marginBottom: StyleSize.Medium }}
      >
        <oh-message body="A" />
        <oh-message body="B" />
      </oh-box>
      <oh-collapse-box open={false} label="More">
        <oh-message body="C" />
      </oh-collapse-box>
      <oh-box
        direction={Direction.Vertical}
        justifyContent={JustifyContent.Left}
        alignItems={AlignItems.Center}
        layout={{ marginTop: StyleSize.Medium, marginBottom: StyleSize.Medium }}
      >
        <oh-message body="D" />
        <oh-message body="E" />
      </oh-box>
    </>
  );
};
```

This isn't too much of a pain, but it would certainly be way easier for me to just be able to have one box instead of multiple. Right now, I'm not looking for any collapse boxes within a collapse box or anything like that. Just the ability to add a collapse box to a parent box.

**Comment by MichaelOliveMcCarrick** [comment]
Hey thanks for providing the example. After talking it over with the team, we've decided that this is going to be a works as designed, with no future plans (currently) to allow CollapseBox nesting within a Box component. It's possible we may revisit this in the future as it's definitely something that seems to keep popping up every now and then.

**Comment by kevinfiol** [comment]
Ran into this issue today. I think it's a bad move to not fix. Boxes essentially serve as our `div` equivalent in our loops since we have no simpler way to group child elements (without using fragments). In addition, boxes provide simple ways to add padding or margins to grouped sections, and the chances don't seem rare that you'd eventually want to add a collapsible box in your tree.

**Comment by MaggieMaSS** [comment]
@kevinfiol Thanks for the update, we're sorry for your inconvenience,  we have brought this up again and will discuss again if there's a better way to track this issue! We'll keep you updated!

**Comment by MaggieMaSS** [comment]
Thanks for the feedback! We have this feature released in LDK 3.15.0. :)
